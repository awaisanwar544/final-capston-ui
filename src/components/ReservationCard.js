function Reservation() {
  return (
    <div className="flex justify-around mt-8 shadow-lg w-3/4 border-4 border-transparent hover:border-red-500 hover:rounded-md hover:shadow-2xl hover:scale-105">
      <div>
        <img className="w-20 h-20 m-auto object-contain rounded-full bg-custom-green-500" src="https://w7.pngwing.com/pngs/905/43/png-transparent-management-professional-sales-business-service-lawyer-company-service-people-thumbnail.png" alt="Person" />
      </div>
      <div className="text-center">
        <p>You reserved Person</p>
        <p>From date to date</p>
        <p className="flex justify-around">
          Total cost:
          <b>$1</b>
        </p>
        <p className="text-custom-grey-500">(reserved on date)</p>
      </div>
    </div>
  );
}

export default Reservation;
