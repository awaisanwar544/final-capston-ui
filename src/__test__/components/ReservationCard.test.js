import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import { Routes, Route } from 'react-router';
import store from '../../redux/store';
import ReservationCard from '../../components/ReservationCard';

describe('App', () => {
  test('Render App', () => {
    localStorage.setItem('user', JSON.stringify({
      name: 'John Doe',
      admin: true,
    }));

    const reservationData = {
      id: 1,
      provider_id: 1,
      start_date: '2019-01-01',
      end_date: '2019-01-02',
      total_cost: '$100',
      created_at: '2019-01-01',
      provider_name: 'John Doe',
      provider_image: 'https://i.imgur.com/w3duR07.jpg',
    };
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" exact element={<ReservationCard reservationData={reservationData} />} />
          </Routes>
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
