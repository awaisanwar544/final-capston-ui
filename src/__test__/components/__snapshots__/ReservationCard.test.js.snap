// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render App 1`] = `
<div
  className="flex justify-around mt-8 shadow-lg w-3/4 border-4 border-transparent hover:shadow-2xl transition hover:scale-110 duration:300ms"
  id="reservation-1"
>
  <div>
    <img
      alt="John Doe"
      className="w-20 h-20 m-auto object-contain rounded-full bg-custom-green-500"
      src="https://i.imgur.com/w3duR07.jpg"
    />
  </div>
  <div
    className="flex items-center"
  >
    <div
      className="text-center"
    >
      <p
        className="flex justify-around"
      >
        You reserved:
        <b>
          John Doe
        </b>
      </p>
      <p>
        From December 01
               to January 02
      </p>
      <p
        className="flex justify-around"
      >
        Total cost:
        <b>
          $100
        </b>
      </p>
      <p
        className="text-custom-grey-500"
      >
        (Reserved on 01 / 01 / 2019)
      </p>
    </div>
  </div>
  <div
    className="flex items-center justify-center"
  >
    <button
      className="bg-custom-white-500 text-custom-grey-500 border border-custom-green-500 hover:bg-red-500 hover:text-white hover:border-red-500 font-bold py-2 px-4 rounded focus:outline-none"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
